inc/functions-shortcodes.php

```diff
@@ -215,3 +218,51 @@ function members_login_form_shortcode() {
 
 	return wp_login_form( array( 'echo' => false ) );
 }
+
+/**
+ * Filters the login redirect URL to send failed logins back to the
+ * referrer with a query arg of `login=failed`.
+ *
+ * @since 3.2.18
+ *
+ * @param string $redirect_to    The redirect destination URL.
+ * @param string $request        The request URL.
+ * @param object $user           The user object.
+ * @return string                The redirect URL.
+ */
+function members_login_redirect( $redirect_to, $request, $user ) {
+    // If there's a valid referrer, and it's not the default log-in screen
+    if (
+        isset( $_SERVER['HTTP_REFERER'] ) &&
+        ! empty( $_SERVER['HTTP_REFERER'] ) &&
+        ( strstr( $_SERVER['HTTP_REFERER'],'wp-login' ) ||
+        strstr( $_SERVER['HTTP_REFERER'],'wp-admin' ) )
+    ) {
+        return $redirect_to;
+    } elseif ( empty( $user ) || is_wp_error( $user ) ) {
+        wp_redirect( str_replace('?login=failed', '', $_SERVER['HTTP_REFERER']) . '?login=failed' );
```

Comment:
@ThemeGravity if I try to visit https://site.com/wp-login.php or https://site.com/wp-admin/ when not logged in. It creates an infinite redirect loop.

I think a better approach here would be to create a hidden input in the members login form and then check if that hidden field exists here before redirecting.